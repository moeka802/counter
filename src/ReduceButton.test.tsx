import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';
import ReduceButton from '../src/ReduceButton';

describe('ReduceButton', () => {
  describe('countが0より大きい時', () => {
    const count = 10;
    it('通常のReduceButtonを描画できる', () => {
      const component = renderer.create(
        <ReduceButton count={count} setCounter={() => {}} />,
      );
      expect(component).toMatchSnapshot();
    });
    it(`ボタン押下でカウントが${count}から${count - 1}になる`, () => {
      const onPressEvent = jest.fn();
      const {container} = render(
        <ReduceButton count={count} setCounter={onPressEvent} />,
      );
      fireEvent.(container.children[0]);
      expect(onPressEvent).toBeCalled();
      expect(onPressEvent.mock.calls[0][0]).toBe(count - 1);
    });
  });
  describe('countが0の時', () => {
    const count = 0;
    it('グレーアウトされたReduceButtonを描画できる', () => {
      const component = renderer.create(
        <ReduceButton count={count} setCounter={() => {}} />,
      );
      expect(component).toMatchSnapshot();
    });
    it('ボタン押下できない', () => {
      const onPressEvent = jest.fn();
      const {container} = render(
        <ReduceButton count={count} setCounter={onPressEvent} />,
      );
      fireEvent.press(container.children[0]);
      expect(onPressEvent).not.toBeCalled();
    });
  });
});
