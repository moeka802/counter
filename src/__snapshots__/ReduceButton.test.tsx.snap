// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReduceButton countが0の時 グレーアウトされたReduceButtonを描画できる 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  disabled={true}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#008080",
        "height": 50,
        "justifyContent": "center",
        "width": 50,
      },
      Object {
        "backgroundColor": "gray",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
      }
    }
  >
    -
  </Text>
</TouchableOpacity>
`;

exports[`ReduceButton countが0より大きい時 通常のReduceButtonを描画できる 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  disabled={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#008080",
        "height": 50,
        "justifyContent": "center",
        "width": 50,
      },
      false,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
      }
    }
  >
    -
  </Text>
</TouchableOpacity>
`;
